rust/connlib/tunnel/src/messages/client.rs

```diff
@@ -328,44 +201,66 @@ mod tests {
-                        .to_owned(),
-                ]),
-            }),
-            Some(OutboundRequestId::for_test(6)),
-        );
-
-        let ingress_message = serde_json::from_str::<PhoenixMessage<_, ()>>(message).unwrap();
-
-        assert_eq!(ingress_message, expected);
+    fn can_deserialize_unknown_resource() {
+        let resources = r#"[
+            {
+                "id": "73037362-715d-4a83-a749-f18eadd970e6",
+                "type": "cidr",
+                "name": "172.172.0.0/16",
+                "address": "172.172.0.0/16",
+                "address_description": "cidr resource",
+                "gateway_groups": [{"name": "test", "id": "bf56f32d-7b2c-4f5d-a784-788977d014a4"}]
+            },
+            {
+                "id": "03000143-e25e-45c7-aafb-144990e57dcd",
+                "type": "dns",
+                "name": "gitlab.mycorp.com",
+                "address": "gitlab.mycorp.com",
+                "address_description": "dns resource",
+                "gateway_groups": [{"name": "test", "id": "bf56f32d-7b2c-4f5d-a784-788977d014a4"}]
+            },
+            {
+                "id": "1106047c-cd5d-4151-b679-96b93da7383b",
+                "type": "internet",
+                "name": "Internet Resource",
+                "gateway_groups": [{"name": "test", "id": "eb94482a-94f4-47cb-8127-14fb3afa5516"}],
+                "not": "relevant",
+                "some_other": [
+                    "field"
+                ]
+            },
+            {
+                "type": "what_is_this",
+            }
+        ]"#;
+
+        serde_json::from_str::<Vec<ResourceDescription>>(resources).unwrap();
     }
 
     #[test]
-    fn invalidate_ice_candidates_message() {
-        let msg = r#"{"event":"invalidate_ice_candidates","ref":null,"topic":"client","payload":{"candidates":["candidate:7854631899965427361 1 udp 1694498559 172.28.0.100 47717 typ srflx"],"gateway_id":"2b1524e6-239e-4570-bc73-70a188e12101"}}"#;
-        let expected = IngressMessages::InvalidateIceCandidates(GatewayIceCandidates {
-            gateway_id: "2b1524e6-239e-4570-bc73-70a188e12101".parse().unwrap(),
-            candidates: vec![
-                "candidate:7854631899965427361 1 udp 1694498559 172.28.0.100 47717 typ srflx"
-                    .to_owned(),
-            ],
-        });
-
-        let actual = serde_json::from_str::<IngressMessages>(msg).unwrap();
-
-        assert_eq!(actual, expected);
+    fn can_deserialize_ice_candidates_message() {
+        let json = r#"{"topic":"client","event":"ice_candidates","payload":{"gateway_ids":["b3d34a15-55ab-40df-994b-a838e75d65d7"],"candidates":["candidate:7031633958891736544 1 udp 50331391 35.244.108.190 53909 typ relay"]},"ref":6}"#;
+
+        let message = serde_json::from_str::<IngressMessages>(json).unwrap();
+
+        assert!(matches!(message, IngressMessages::IceCandidates(_)));
```

Comment:
Being able to deserialize into the right variant is a good-enough test IMO. The value of these tests is kind of low anyway because there is no guarantee that the portal sends the messages like this.

If the rest of the code compiles because we can get the right data out of the message types, then knowing that we are deserialising into the right type is good enough and saves us lots of typing here.